python/example_load_gen.py

```diff
@@ -0,0 +1,307 @@
+            context.get_latencies.record_value(get_duration)
+            if get_result.status() == CacheGetStatus.HIT:
+                value = get_result.value()
+                value_string = f"{value[0:10]}... (len: {len(value)})"
+            else:
+                value_string = 'n/a'
+
+            if context.global_request_count % 1_000 == 0:
+                self.logger.info(
+                    f"worker: {worker_id}, worker request: {operation_id}, global request: {context.global_request_count}, status: {get_result.status()}, val: {value_string}"
+                )
+
+    T = TypeVar("T")
+
+    async def execute_request_and_update_context_counts(
+            self,
+            context: BasicPythonLoadGenContext,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Optional[T]:
+        result, response = await self.execute_request(block)
+        self.update_context_counts_for_request(context, result)
+        return response
+
+    async def execute_request(
+            self,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Tuple[AsyncSetGetResult, Optional[T]]:
+        try:
+            result = await block()
+            return AsyncSetGetResult.SUCCESS, result
+        except InternalServerError as e:
+            # TODO verify exception type
+            self.logger.error(f"Caught InternalServerError: {e}")
+            return AsyncSetGetResult.UNAVAILABLE, None
+        except TimeoutError as e:
+            # TODO need to verify exception type
+            self.logger.error(f"Caught TimeoutError: {e}")
+            return AsyncSetGetResult.DEADLINE_EXCEEDED, None
+
+    @staticmethod
+    def update_context_counts_for_request(
+            context: BasicPythonLoadGenContext,
+            result: AsyncSetGetResult
+    ):
+        context.global_request_count += 1
+        if result == AsyncSetGetResult.SUCCESS:
+            context.global_success_count += 1
+        elif result == AsyncSetGetResult.UNAVAILABLE:
+            context.global_unavailable_count += 1
+        elif result == AsyncSetGetResult.DEADLINE_EXCEEDED:
+            context.global_deadline_exceeded_count += 1
+        else:
+            raise ValueError(f"Unsupported result type: {result}")
+
+    def tps(self, context: BasicPythonLoadGenContext, request_count: int) -> int:
+        return round((request_count * 1000) / self.get_elapsed_millis(context.start_time))
+
+    @staticmethod
+    def percent_requests(context: BasicPythonLoadGenContext, count: int) -> float:
+        return round((count / context.global_request_count) * 100 * 10) / 10
+
+    @staticmethod
+    def output_histogram_summary(
+            histogram: HdrHistogram
+    ) -> str:
+        return f"""
+    count: {histogram.total_count}
+      min: {histogram.min_value}
+      p50: {histogram.get_value_at_percentile(50)}
+      p90: {histogram.get_value_at_percentile(90)}
+      p99: {histogram.get_value_at_percentile(99)}
```

Comment:
maybe add p999 isnt that what our SLA is benched at?

Reply:
discussed in slack leaving as is in pr for now.

Reply:
2 9's imo is enough for interpreted languages - particularly when you expect them to be run on a laptop over wifi.